autoPyTorch/datasets/base_dataset.py

```diff
@@ -97,18 +97,15 @@ def __init__(
             val_transforms (Optional[torchvision.transforms.Compose]):
                 Additional Transforms to be applied to the validation/test data
         """
-        if dataset_name is not None:
-            self.dataset_name = dataset_name
-        else:
-            self.dataset_name = hash_array_or_matrix(train_tensors[0])
+        self.dataset_name = dataset_name if dataset_name is not None \
```

Comment:
I think is better to prevent the use of `\` when possible. 

Also, this `hash_array_or_matrix` function is something we do not user anymore... maybe try using [this](https://github.com/automl/Auto-PyTorch/blob/b3c301c66c230a5a4040ca1fc1a20858d2e2d9dc/autoPyTorch/api/base_task.py#L914).

Reply:
Ok, so 
`self.dataset_name = str(uuid.uuid1(clock_seq=os.getpid()))`?